### PR TITLE
Invalid [resistance] who don't have max_value attribute

### PR DESCRIPTION
**Pull request by newfrenchy83**
Fix https://github.com/wesnoth/wesnoth/issues/8092 issue.

If [resistance] ability don't contain "max_value" attribute, it will be removed of the list because invalid

### CODE REVIEW

**Comment by ProditorMagnus** [comment]
I dont agree with this. Current default max_value=0 removing positive resistances is example how [resistance] functions.

It would be common-use-case-friendly to increase default max_value from 0 to 100 though.

File: src/units/unit.cpp [hunk 1]
```diff
@@ -1792,7 +1792,7 @@ int unit::resistance_against(const std::string& damage_name,bool attacker,const
 
 	unit_ability_list resistance_abilities = get_abilities_weapons("resistance",loc, weapon, opp_weapon);
 	utils::erase_if(resistance_abilities, [&](const unit_ability& i) {
-		return !resistance_filter_matches(*i.ability_cfg, attacker, damage_name, 100-res);
+		return ((*i.ability_cfg)["max_value"].empty() || !resistance_filter_matches(*i.ability_cfg, attacker, damage_name, 100-res));
```
[the remark this answers is not in the document]
Comment (newfrenchy83) [reply src/units/unit.cpp:1795]: If many [resistance] abilitie in the list don't have max_value, error message repeated many time, i prefer don't push here or then make trigering one time for all.

**Comment by newfrenchy83** [comment]
> I'm fine with having it default to 100.

I prefer not, because forget max_value incode should concern what ability and not all list. Invalidate ability who don't have code is more selective, else, if an [resistance] have max_value but not another, the two work with max_value less great, when i write this i think what choice between max_value and value, add,etc attribute must be done element by element and not for list but for don't have a debate to that i remain on invalidation f element not conform

File: src/units/unit.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (newfrenchy83) [reply src/units/unit.cpp:1800]: error massage added

**Comment by Pentarctagon** [comment]
This isn't urgent, so I think it'll be good to leave this open for a bit to see if anyone else has opinions on it.

**Comment by Pentarctagon** [comment]
> I expect addon developer added note about max_value being required because their use case assumed that default was 100.

Looks that was it: https://wiki.wesnoth.org/index.php?title=AbilitiesWML&oldid=30221

> Had a fun 1.5 hours trying to figure out why [resistance] wasn't working.

File: src/units/unit.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (newfrenchy83) [reply src/units/unit.cpp:1800]: test work finally

[the remark this answers is not in the document]
Comment (newfrenchy83) [reply src/units/unit.cpp:1795]: message error removed for don't interrupt play

**Pull request closed without merging**